### ISSUE TITLE
Bug report: path fill bug when points value < 0

### ISSUE THREAD

**Issue by akira-cn** [pshihn/rough#139]
Rough fill seems to ignore the path coordinate < 0.

So the following code only fill half of the triangle.

```js
const rc = rough.canvas(document.querySelector('canvas'));
const ctx = rc.ctx;
ctx.translate(150, 150);
ctx.scale(1, -1);
rc.path('M-100, 0L0 100L100 100Z', {fill: 'red'});
```

![](https://p4.ssl.qhimg.com/t012ea7b6ec8022acb0.jpg)

**Comment by pshihn** [comment]
I suspect it only happens when using SVG Paths and not when drawing a polygon.

**Comment by akira-cn** [comment]
Yes, it only happens when using SVG Paths in canvas 2d context.

**Comment by pshihn** [comment]
Okay I am working on a new algorithm to fill arbitrary paths. Should resolve this in a couple of days .

**Comment by pshihn** [comment]
Thanks for reporting this.

**Comment by pshihn** [comment]
This has been fixed in v4.2.0
